lib/cog.ex

```diff
@@ -41,55 +44,65 @@ defmodule Cog do
   @doc "The name of the site namespace."
   def site_namespace, do: "site"
 
+  ########################################################################
+  # Adapter Resolution Functions
+
   @doc """
   Returns the currently configured chat adapter module, if found.
   """
-  @spec adapter_module :: {:ok, module} | {:error, {:bad_adapter, String.t}}
-  def adapter_module do
-    configured = Application.get_env(:cog, :adapter)
-    case adapter_module(configured) do
-      {:ok, adapter} ->
-        {:ok, adapter}
-      {:error, {:bad_adapter, _}} ->
-        {:error, {:bad_adapter, configured}}
-    end
-  end
+  @spec chat_adapter_module :: {:ok, module} | {:error, {:bad_adapter, configured_name :: String.t}}
+  def chat_adapter_module,
+    do: chat_adapter_module(Application.get_env(:cog, :adapter))
 
   @doc """
-  For a given adapter name return the implementing module, if it
-  exists.
+  For a given _chat_ adapter name return the implementing module, if
+  it exists.
   """
-  @spec adapter_module(String.t) :: {:ok, module} | {:error, {:bad_adapter, String.t}}
-  def adapter_module(name) do
-    case Map.fetch(@adapters, name) do
+  @spec chat_adapter_module(String.t) :: {:ok, module} | {:error, {:bad_adapter, String.t}}
+  def chat_adapter_module(name),
+    do: adapter_module(name, chat_adapters)
+
+  @doc """
+  Same as `chat_adapter_module/1` but for _any_ adapter, chat or
+  otherwise.
+  """
+  @spec chat_adapter_module(String.t) :: {:ok, module} | {:error, {:bad_adapter, String.t}}
+  def adapter_module(name),
+    do: adapter_module(name, all_adapters)
+
+  # Note: chat_adapters/0, non_chat_adapters/0, and all_adapters/0 are
+  # only really public to facilitate testing; once we have a more
+  # dynamic scheme for managing adapters, they won't be necessary.
+
+  # TODO: the "null" and "test" adapters need to be brought in only in
+  # non-prod environments
+  def chat_adapters do
+    %{"slack"   => Cog.Adapters.Slack,
+      "hipchat" => Cog.Adapters.HipChat,
+      "irc"     => Cog.Adapters.IRC,
+      "null"    => Cog.Adapters.Null,
+      "test"    => Cog.Adapters.Test}
+  end
```

Comment:
I wish this was more data driven. Not suggesting you change it now btw. It just bothers me for some reason to have the adapters hard coded.